app/src/main/java/com/github/fribourgsdp/radio/MainActivity.kt

```diff
@@ -18,23 +21,42 @@ import com.github.fribourgsdp.radio.deprecated.VoiceOverIPActivity
 import com.github.fribourgsdp.radio.game.prep.GameSettingsActivity
 import com.github.fribourgsdp.radio.game.prep.JoinGameActivity
 
-class MainActivity : MyAppCompatActivity() {
+open class MainActivity : MyAppCompatActivity() {
     private val db = FirestoreDatabase()
+    private lateinit var playButton : Button
+    private lateinit var joinButton : Button
+    private lateinit var settingsButton : Button
+    private lateinit var profileButton : ImageButton
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-
-        val playButton = findViewById<Button>(R.id.playButton)
-        playButton.setOnClickListener {startActivity(Intent(this, GameSettingsActivity::class.java))}
-        val joinButton : Button = findViewById(R.id.joinButton)
-        joinButton.setOnClickListener {startActivity(Intent(this, JoinGameActivity::class.java))}
-        val settingsButton = findViewById<Button>(R.id.settingsButton)
+        initViews()
+        playButton.setOnClickListener {
+            if(hasConnectivity(this)){
+                startActivity(Intent(this, GameSettingsActivity::class.java))
+            }
+            else{
+                Toast.makeText(this,getString(R.string.offline_error_message_toast), Toast.LENGTH_SHORT).show()
+                disableButtons()
+            }
+        }
+        joinButton.setOnClickListener {
+            if(hasConnectivity(this)){
+                startActivity(Intent(this, JoinGameActivity::class.java))
+            }else{
+                Toast.makeText(this,getString(R.string.offline_error_message_toast), Toast.LENGTH_SHORT).show()
+                disableButtons()
+            }
+        }
         settingsButton.setOnClickListener {startActivity(Intent(this, SettingsActivity::class.java))}
-        val profileButton: ImageButton = findViewById(R.id.profileButton)
         profileButton.setOnClickListener {
             startActivity(Intent(this, UserProfileActivity::class.java))
         }
 
+        if(!hasConnectivity(this)) {
+            disableButtons()
+        }
```

Comment:
If this is called before the users click on the play or join button, both of these will be disabled and the user may never understand why they cannot play. Could you maybe add the same Toast here to explain why the buttons are disabled? (i.e add the Toast in the disableButtons method and maybe rename it)

Reply:
The idea would be to display the toast when loading the page?

Reply:
Yes to explain to the user why they cannot do anything.